frontend/src/js/controllers/analyticsCtrl.js

```diff
@@ -0,0 +1,296 @@
+        parameters.url = 'hosts/challenge_host_team/';
+        parameters.method = 'GET';
+        parameters.token = userKey;
+        parameters.callback = {
+            onSuccess: function(response) {
+                var status = response.status;
+                var details = response.data;
+                if (status == 200) {
+                    vm.hostTeam = details.results;
+
+                }
+            },
+            onError: function(response) {
+                var status = response.status;
+                var error = response.data;
+                if (status == 403) {
+                    vm.error = error;
+
+                    // navigate to permissions denied page
+                    $state.go('web.permission-denied');
+                } else if (status == 401) {
+                    alert("Timeout, Please login again to continue!");
+                    utilities.resetStorage();
+                    $state.go("auth.login");
+                    $rootScope.isAuth = false;
+
+                }
+            }
+        };
+        utilities.sendRequest(parameters);
+
+
+        parameters.url = 'challenges/challenge?mode=host';
+        parameters.method = 'GET';
+        parameters.token = userKey;
+        parameters.callback = {
+            onSuccess: function(response) {
+                var status = response.status;
+                var details = response.data;
+                if (status == 200) {
+                    vm.challengeList = details.results;
+                    vm.challengeListCount = details.count;
+                }
+            },
+            onError: function(response) {
+                var status = response.status;
+                var error = response.data;
+                if (status == 403) {
+                    vm.error = error;
+
+                    // navigate to permissions denied page
+                    $state.go('web.permission-denied');
+                } else if (status == 401) {
+                    alert("Timeout, Please login again to continue!");
+                    utilities.resetStorage();
+                    $state.go("auth.login");
+                    $rootScope.isAuth = false;
+
+                }
+            }
+        };
+        utilities.sendRequest(parameters);
+
+        vm.showChallengeAnalysis = function() {
+            if (vm.challengeId != null) {
+                parameters.url = 'challenges/challenge/' + vm.challengeId + '/challenge_phase';
+                parameters.method = 'GET';
+                parameters.token = userKey;
+                parameters.callback = {
+                    onSuccess: function(response) {
+                        var status = response.status;
+                        var details = response.data;
+
+
+                        if (status === 200) {
+                            vm.currentPhase = details.results;
+
+                            for (var phaseCount = 0; phaseCount < vm.currentPhase.length; phaseCount++) {
```

Comment:
https://stackoverflow.com/questions/500504/why-is-using-for-in-with-array-iteration-a-bad-idea
Check this please.